homeassistant/components/zerproc/light.py

```diff
@@ -110,17 +110,16 @@ async def async_added_to_hass(self) -> None:
         """Run when entity about to be added to hass."""
         self.async_on_remove(
             self.hass.bus.async_listen_once(
-                EVENT_HOMEASSISTANT_STOP, self.on_hass_shutdown
+                EVENT_HOMEASSISTANT_STOP, self.async_will_remove_from_hass
             )
         )
 
-    async def async_will_remove_from_hass(self) -> None:
+    async def async_will_remove_from_hass(self, *args) -> None:
         """Run when entity will be removed from hass."""
-        await self._light.disconnect()
-
-    async def on_hass_shutdown(self, event):
-        """Execute when Home Assistant is shutting down."""
-        await self._light.disconnect()
+        try:
+            await self._light.disconnect()
+        except pyzerproc.ZerprocException:
+            pass
```

Comment:
One final small concern I, should such an occurrence be logged? (e.g., as a warning, or maybe even as info/debug ?)

Reply:
:+1: I'll add a debug log for it. The assumption here is that if there's an exception disconnecting, the connection was probably already closed.